com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/scoping/AbstractNameFunction.java

```diff
@@ -45,7 +49,8 @@ public AbstractNameFunction(final String representation) {
   /** {@inheritDoc} */
   @Override
   public QualifiedName apply(final IEObjectDescription from) {
-    return apply(from.getEObjectOrProxy());
+    LOG.warn("No explicit name function for description " + from.getEObjectURI() + " of type " + EcoreUtil.getURI(from.getEClass())); //$NON-NLS-1$ //$NON-NLS-2$

```

Comment:
It would be nice if we could make sure to never get the warning message. I suppose that's the next step with the split of naming functions. Practically if this warning occurs on a large scope the builder will probably end up doing nothing, but writing this into the log.

Reply:
Yes, the plan was to only leave this in for a transition period until all downstream projects have adapted to these changes.